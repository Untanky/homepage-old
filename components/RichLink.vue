<template lang="pug">
  a.flex.flex-col.p-4.items-center.justify-between.group(:href="linkUrl" :title="linkAltText")
    img.w-16.h-16(:src="imageUrl" :alt="linkAltText")
    .text-xl {{title}}
    div(class="after:chevron") {{text}}
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'RichLink',
  props: {
    title: String,
    text: String,
    imageUrl: String,
    linkUrl: String,
  },
  computed: {
    linkAltText(): string {
      return `${this.title} - ${this.text}`;
    },
  },
});
</script>
