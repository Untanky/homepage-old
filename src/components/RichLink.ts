import ComponentData from "../interfaces/ComponentData";

interface RichLink extends ComponentData {
  title: string,
  text: string,
  imageUrl: string,
  linkUrl: string,
  classes: string[],
}

export default RichLink;
