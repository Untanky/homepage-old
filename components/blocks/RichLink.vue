<template>
  <a
    class="flex flex-col p-4 items-center justify-between group"
    :class="data.classes"
    :href="data.linkUrl"
    :title="linkAltText"
  >
    <img
      class="w-16 h-16"
      :src="data.imageUrl"
      :alt="linkAltText"
    />
    <div
      class="text-xl"
    >
      {{data.title}}
    </div>
    <div
      class="link after:chevron group-hover:underline"
    >
      {{data.text}}
    </div>
  </a>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import RichLink from '../../src/components/RichLink';

export default Vue.extend({
  name: 'RichLink',
  props: {
    data: Object as PropType<RichLink>,
  },
  computed: {
    linkAltText(): string {
      return `${this.data.title} - ${this.data.text}`;
    },
  },
});
</script>
