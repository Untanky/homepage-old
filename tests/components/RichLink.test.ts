/* eslint-disable no-undef */
// global it, expect
import { mount } from '@vue/test-utils';
import RichLink from '../../components/RichLink.vue';

it('should run', () => {
  const props = {
    title: 'Title',
    text: 'Link Text',
    linkUrl: 'https://example.com',
    imageUrl: 'https://example.com/img.jpg',
  };

  const comp = mount(RichLink, { propsData: props });
  expect(comp.element).toMatchSnapshot();
});
