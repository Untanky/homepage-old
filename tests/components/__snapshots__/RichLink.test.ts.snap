// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should run 1`] = `
<a
  class="flex flex-col p-4 items-center justify-between group"
  href="https://example.com"
  title="Title - Link Text"
>
  <img
    alt="Title - Link Text"
    class="w-16 h-16"
    src="https://example.com/img.jpg"
  />
  <div
    class="text-xl"
  >
    Title
  </div>
  <div
    class="after:chevron"
  >
    Link Text
  </div>
</a>
`;
